### DISCUSSION TITLE
v1.7.6

### DISCUSSION THREAD

**Discussion by FenPhoenix** [FenPhoenix/AngelLoader#116] [General]
Features and improvements:
- Added support for The Dark Mod. As part of this:
  - Backup path is now only required if you've specified at least one game that uses it. The Dark Mod doesn't use it; all other games do.
  - For Dark Mod FMs, the "installed" icon means that the FM is TDM's "selected" one (the one that loads up when you start the game), and only one Dark Mod FM at a time can be so marked. Thus, for Dark Mod FMs, "Install FM" / "Uninstall FM" becomes "Select FM" / "Deselect FM". The "selected" FM is also kept in sync with the game, if you happen to select a different one there.
  - The FMs list is automatically refreshed whenever you download a new FM in-game.
  - For The Dark Mod only, the finished-on difficulty will be automatically updated when you finish an FM; there is no need to set it manually.
  - Dark Mod FM scans use a mix of local data and data pulled from The Dark Mod's server. This allows for more accuracy than would be possible with a local-only scan. If it can't connect to the TDM server, it falls back to a local-only scan.

- If the backup path is not specified or doesn't exist, the DarkLoader Import window will now disable the Import Saves checkbox and provide an option to set the backup path.

- Improved the scanner's non-acronym title selector even more.
- Improved the scanner's cleanup of titles/authors a bit.

- In Settings -> Appearance -> Show or hide interface elements, the "Hide" checkboxes are now "Show" checkboxes. This removes a double negative type thing and makes it more straightforward.

- Each game now has its own customizable web search URL, with sensible defaults.
- Web Search option is now disabled for unknown/unsupported FMs. The option didn't do anything for these FMs anyway, so this just makes that clear.

Performance:
- Greatly improved rtf parser performance.

Fixes:
- If there were new FMs to be scanned on startup, the list would be sorted BEFORE the scan, meaning if the scanner gave the FMs titles that didn't sort the same as their placeholder archive name based titles, the list would not be correctly sorted until the user chose to sort it again. This bug had been in there literally forever. :(
- Fixed encoding detection for "Nightwalk" and "Troubling Transitions" readmes.
- In the Settings window, if a setting was changed that required a view refresh (game paths etc.), the filters would lose their state after the refresh.
- If an FM with an unknown or unsupported game type is selected, the messages in the tab area now no longer refer to Thief 3 ("Mod management is not supported for Thief: Deadly Shadows").

<hr /><em>This discussion was created from the release <a href='https://github.com/FenPhoenix/AngelLoader/releases/tag/v1.7.6'>v1.7.6</a>.</em>

**Comment by jonathanlinat** [comment]
Nice! Quick question. As you added TDM support, shouldn't that be defined as version `1.8.0` instead of `1.7.6`? It's a pretty big achievement, after all.

**Reply by FenPhoenix** [reply]
Eh... maybe. I've bumped the version like that before for major updates, but it's too late now. Kinda arbitrary anyway, and what if I ran up to 1.9.9 and the next update wasn't major? 2.0 would just be like a bugfix release, so that wouldn't be any better. ¯\\\_(ツ)\_/¯

**Reply by jonathanlinat** [reply]
You can do `1.10.0` in that case. You are not limited to the first nine numbers. It was just a [semver](https://semver.org/) suggestion. Anyway, that's okay 👍🏻

**Reply by FenPhoenix** [reply]
Oh yeah 😅 Well maybe it can be a humble brag then, "meh, adding Dark Mod was no sweat. Not big enough for a version bump, easy stuff, easy stuff."
